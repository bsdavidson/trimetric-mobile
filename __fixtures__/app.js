export const props = {
  dimensions: {
    screen: {
      width: 500,
      height: 600
    }
  },
  following: false,
  layerVisibility: {},
  loaded: true,
  routeShapes: [],
  selectedArrival: {},
  selectedArrivalVehicleInfo: {},
  selectedItem: {},
  selectedItemsInfo: [],
  selectedItemsViewHeight: 100,
  selectedVehicleInfo: {},
  stopPoints: [],
  vehiclePoints: []
};
