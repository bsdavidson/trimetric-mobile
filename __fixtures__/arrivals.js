export const selectedStop = {
  type: "stop",
  position: [-122.68931172788143, 45.51957800931885],
  item: {
    type: "Feature",
    properties: {stop_id: "5016", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68931172788143, 45.51957800931885]
    }
  }
};

export const selectedItem = {
  type: "stop",
  position: [-122.68931172788143, 45.51957800931885],
  item: {
    type: "Feature",
    properties: {stop_id: "5016", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68931172788143, 45.51957800931885]
    }
  }
};

export const arrivals = [
  {
    route_id: "51",
    route_short_name: "51",
    route_long_name: "Vista",
    route_type: 3,
    route_color: "",
    route_text_color: "",
    trip_id: "7908833",
    stop_id: "5016",
    headsign: "Portland",
    arrival_time: "16:22:00",
    departure_time: "16:22:00",
    vehicle_id: null,
    vehicle_label: null,
    vehicle_position: {
      lat: 0,
      lng: 0,
      bearing: 0,
      odometer: 0,
      speed: 0
    },
    date: "2018-01-19T00:00:00Z",
    trip_shape: {
      direction_id: 1,
      route_id: "51",
      color: "",
      points: [
        {lat: 45.493347, lng: -122.70053},
        {lat: 45.493347, lng: -122.700701},
        {lat: 45.516844, lng: -122.678826}
      ],
      trip_id: "7908833"
    }
  },
  {
    route_id: "51",
    route_short_name: "51",
    route_long_name: "Vista",
    route_type: 3,
    route_color: "",
    route_text_color: "",
    trip_id: "7908834",
    stop_id: "5016",
    headsign: "Portland",
    arrival_time: "16:52:00",
    departure_time: "16:52:00",
    vehicle_id: null,
    vehicle_label: null,
    vehicle_position: {
      lat: 0,
      lng: 0,
      bearing: 0,
      odometer: 0,
      speed: 0
    },
    date: "2018-01-19T00:00:00Z",
    trip_shape: {
      direction_id: 1,
      route_id: "51",
      color: "",
      points: [
        {lat: 45.490581, lng: -122.709817},
        {lat: 45.490589, lng: -122.710375},
        {lat: 45.516844, lng: -122.678826}
      ],
      trip_id: "7908834"
    }
  }
];

export const selectedItems = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      properties: {stop_id: "5016", type: "stop"},
      geometry: {
        type: "Point",
        coordinates: [-122.68931172788143, 45.51957800931885]
      }
    }
  ]
};
