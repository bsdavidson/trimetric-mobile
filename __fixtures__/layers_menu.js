export const display = true;

export const bottom = 60;

export const selectedItemsViewHeight = 0;

export const layerVisibility = {
  routeShapes: true,
  buses: false,
  trains: true,
  stops: true,
  vehicleLabels: true,
  buildings: false
};

export const dimensions = {
  window: {fontScale: 1, width: 375, height: 667, scale: 2},
  screen: {fontScale: 1, width: 375, height: 667, scale: 2}
};
