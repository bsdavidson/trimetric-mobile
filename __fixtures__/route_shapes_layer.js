export const props = {
  visible: true,
  routeShapes: {
    type: "FeatureCollection",
    features: [
      {
        type: "Feature",
        properties: {color: "#CE0F69", route_id: "194"},
        geometry: {
          type: "MultiLineString",
          coordinates: [
            [
              [-122.680501, 45.53012],
              [-122.681354, 45.530079],
              [-122.681105, 45.530084]
            ]
          ]
        }
      },
      {
        type: "Feature",
        properties: {color: "#84BD00", route_id: "193"},
        geometry: {
          type: "MultiLineString",
          coordinates: [
            [
              [-122.671375, 45.493955],
              [-122.671107, 45.49395],
              [-122.698618, 45.529861],
              [-122.698646, 45.530509],
              [-122.698649, 45.530616]
            ],
            [
              [-122.685036, 45.514564],
              [-122.685257, 45.514157],
              [-122.685323, 45.514034],
              [-122.685339, 45.514005]
            ]
          ]
        }
      }
    ]
  },
  selectedArrival: null
};
