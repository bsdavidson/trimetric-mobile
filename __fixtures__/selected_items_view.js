// Selected Stop
export const stopData = [
  {
    type: "stop",
    stop: {
      id: "7749",
      code: "7749",
      name: "SW 6th & Broadway",
      desc: "Northbound stop in Portland (Stop ID 7749)",
      lat: 45.506859,
      lng: -122.684151,
      zone_id: "B",
      url: "http://trimet.org/#tracker/stop/7749",
      location_type: 0,
      parent_station: "",
      direction: "North",
      position: "Nearside",
      wheelchair_boarding: 0,
      distance: 0
    },
    arrivals: [
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902197",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:12:07",
        departure_time: "14:12:07",
        vehicle_id: "2652",
        vehicle_label: "8  To NE Dekum",
        vehicle_position: {
          lat: 45.498703,
          lng: -122.68535,
          bearing: 224,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902197"
        },
        nextArrivals: [
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902198",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:22:07",
            departure_time: "14:22:07",
            vehicle_id: "3142",
            vehicle_label: "8  To NE Dekum",
            vehicle_position: {
              lat: 45.497105,
              lng: -122.68317,
              bearing: 208,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.496432, lng: -122.682258},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902198"
            }
          },
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902199",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:32:07",
            departure_time: "14:32:07",
            vehicle_id: null,
            vehicle_label: null,
            vehicle_position: {
              lat: 0,
              lng: 0,
              bearing: 0,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.499398, lng: -122.685589},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902199"
            }
          },
          {
            route_id: "8",
            route_short_name: "8",
            route_long_name: "Jackson Park/NE 15th",
            route_type: 3,
            route_color: "",
            route_text_color: "",
            trip_id: "7902200",
            stop_id: "7749",
            headsign: "NE Dekum",
            arrival_time: "14:42:44",
            departure_time: "14:42:44",
            vehicle_id: null,
            vehicle_label: null,
            vehicle_position: {
              lat: 0,
              lng: 0,
              bearing: 0,
              odometer: 0,
              speed: 0
            },
            date: "2018-01-22T00:00:00Z",
            trip_shape: {
              direction_id: 1,
              route_id: "8",
              color: "",
              points: [
                {lat: 45.499398, lng: -122.685589},
                {lat: 45.571787, lng: -122.660853}
              ],
              trip_id: "7902200"
            }
          }
        ]
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902198",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:22:07",
        departure_time: "14:22:07",
        vehicle_id: "3142",
        vehicle_label: "8  To NE Dekum",
        vehicle_position: {
          lat: 45.497105,
          lng: -122.68317,
          bearing: 208,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.496432, lng: -122.682258},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902198"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902199",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:32:07",
        departure_time: "14:32:07",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902199"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902200",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:42:44",
        departure_time: "14:42:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902200"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902201",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "14:52:44",
        departure_time: "14:52:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.496432, lng: -122.682258},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902201"
        }
      },
      {
        route_id: "8",
        route_short_name: "8",
        route_long_name: "Jackson Park/NE 15th",
        route_type: 3,
        route_color: "",
        route_text_color: "",
        trip_id: "7902202",
        stop_id: "7749",
        headsign: "NE Dekum",
        arrival_time: "15:02:44",
        departure_time: "15:02:44",
        vehicle_id: null,
        vehicle_label: null,
        vehicle_position: {
          lat: 0,
          lng: 0,
          bearing: 0,
          odometer: 0,
          speed: 0
        },
        date: "2018-01-22T00:00:00Z",
        trip_shape: {
          direction_id: 1,
          route_id: "8",
          color: "",
          points: [
            {lat: 45.499398, lng: -122.685589},
            {lat: 45.571787, lng: -122.660853}
          ],
          trip_id: "7902202"
        }
      }
    ]
  }
];

export const following = true;
export const selectedItem = {
  type: "stop",
  position: [-122.68415115773678, 45.50685913695389],
  item: {
    type: "Feature",
    properties: {stop_id: "7749", type: "stop"},
    geometry: {
      type: "Point",
      coordinates: [-122.68415115773678, 45.50685913695389]
    }
  }
};

export const selectedItemIndex = 0;
export const selectedArrival = null;

export const selectedArrivalVehicleInfo = null;
export const selectedStopInfo = {
  id: "7749",
  code: "7749",
  name: "SW 6th & Broadway",
  desc: "Northbound stop in Portland (Stop ID 7749)",
  lat: 45.506859,
  lng: -122.684151,
  zone_id: "B",
  url: "http://trimet.org/#tracker/stop/7749",
  location_type: 0,
  parent_station: "",
  direction: "North",
  position: "Nearside",
  wheelchair_boarding: 0,
  distance: 0
};

// Selected Vehicle
export const vehicleData = [
  {
    type: "vehicle",
    vehicle: {
      trip: {trip_id: "7918798", route_id: "200"},
      vehicle: {id: "206", label: "Green Line to Clackamas"},
      position: {
        lat: 45.50877,
        lng: -122.68381,
        bearing: 331,
        odometer: 0,
        speed: 0
      },
      current_stop_sequence: 1,
      stop_id: "13140",
      current_status: 1,
      timestamp: 1516660571,
      congestion_level: 0,
      occupancy_status: 0,
      route_type: 0
    }
  }
];

export const selectedItemVehicle = {
  type: "vehicle",
  position: [-122.68381, 45.50877],
  item: {
    type: "Feature",
    properties: {
      bearing: 331,
      route_id: "200",
      icon: "tram",
      vehicle_id: "206",
      type: "vehicle",
      stop_id: "13140"
    },
    geometry: {type: "Point", coordinates: [-122.68381, 45.50877]}
  }
};

export const selectedVehicleInfo = {
  id: "13140",
  code: "13140",
  name: "Jackson St Turnaround",
  desc: "Westbound stop in Portland (Stop ID 13140)",
  lat: 45.508653,
  lng: -122.6836,
  zone_id: "R",
  url: "http://trimet.org/#tracker/stop/13140",
  location_type: 0,
  parent_station: "",
  direction: "West",
  position: "At",
  wheelchair_boarding: 0,
  distance: 0
};
