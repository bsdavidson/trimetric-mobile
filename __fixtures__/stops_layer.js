export const stopPoints = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      properties: {type: "stop", stop_id: "13990"},
      geometry: {type: "Point", coordinates: [-122.398702, 45.552013]}
    },
    {
      type: "Feature",
      properties: {type: "stop", stop_id: "13991"},
      geometry: {type: "Point", coordinates: [-122.407683, 45.55363]}
    }
  ]
};

export const stops = [
  {
    id: "2",
    code: "2",
    name: "A Ave & Chandler",
    desc: "Eastbound stop in Lake Oswego (Stop ID 2)",
    lat: 45.420609,
    lng: -122.675671,
    zone_id: "B",
    url: "http://trimet.org/#tracker/stop/2",
    location_type: 0,
    parent_station: "",
    direction: "East",
    position: "Nearside",
    wheelchair_boarding: 0,
    distance: 0
  },
  {
    id: "landmark-2560",
    code: "",
    name: "SE Park Ave MAX Station",
    desc: "",
    lat: 45.430735,
    lng: -122.635065,
    zone_id: "",
    url: "",
    location_type: 1,
    parent_station: "",
    direction: "",
    position: "",
    wheelchair_boarding: 0,
    distance: 0
  }
];
