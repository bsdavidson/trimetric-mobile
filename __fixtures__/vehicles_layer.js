export const props = {
  filter: ["!=", "non_existing_attribute", "1"],
  vehiclePoints: {
    type: "FeatureCollection",
    features: [
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "101",
          route_id: "200",
          stop_id: "7601",
          bearing: 272
        },
        geometry: {type: "Point", coordinates: [-122.67175, 45.526817]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "536",
          route_id: "200",
          stop_id: "8372",
          bearing: 298
        },
        geometry: {type: "Point", coordinates: [-122.579895, 45.534035]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "537",
          route_id: "200",
          stop_id: "8347",
          bearing: 165
        },
        geometry: {type: "Point", coordinates: [-122.563545, 45.53022]}
      },
      {
        type: "Feature",
        properties: {
          type: "vehicle",
          icon: "tram",
          vehicle_id: "538",
          route_id: "90",
          stop_id: "10579",
          bearing: 145
        },
        geometry: {type: "Point", coordinates: [-122.59301, 45.58756]}
      }
    ]
  },
  labelsVisible: true
};

export const vehicles = [
  {
    trip: {trip_id: "7918797", route_id: "200"},
    vehicle: {id: "101", label: "Green Line to Clackamas"},
    position: {
      lat: 45.50887,
      lng: -122.68386,
      bearing: 347,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 1,
    stop_id: "13140",
    current_status: 1,
    timestamp: 1516746492,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  },

  {
    trip: {trip_id: "7919096", route_id: "200"},
    vehicle: {id: "537", label: "Green Line to City Ctr"},
    position: {
      lat: 45.435745,
      lng: -122.56782,
      bearing: 350,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 1,
    stop_id: "13132",
    current_status: 1,
    timestamp: 1516746481,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  },

  {
    trip: {trip_id: "7914971", route_id: "90"},
    vehicle: {id: "538", label: "Red Line to Beaverton"},
    position: {
      lat: 45.53185,
      lng: -122.56438,
      bearing: 292,
      odometer: 0,
      speed: 0
    },
    current_stop_sequence: 7,
    stop_id: "8371",
    current_status: 2,
    timestamp: 1516746477,
    congestion_level: 0,
    occupancy_status: 0,
    route_type: 0
  }
];
