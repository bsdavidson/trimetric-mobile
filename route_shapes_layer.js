import React, {Component} from "react";
import {Platform, StyleSheet, Text, View} from "react-native";
import Mapbox from "@mapbox/react-native-mapbox-gl";
import {connect} from "react-redux";

import {MIN_LABEL_LAYER_ID, EXCLUDE_ALL, INCLUDE_ALL} from "./App";

function RouteShapesLayer(props) {
  const {routeShapes, visible} = props;
  if (!routeShapes) {
    return null;
  }
  return (
    <View>
      <Mapbox.ShapeSource id="route_shapes_source" shape={routeShapes}>
        <Mapbox.LineLayer
          id="route_shapes_layer"
          belowLayerID={MIN_LABEL_LAYER_ID}
          filter={visible ? INCLUDE_ALL : EXCLUDE_ALL}
          style={mapStyles.routeShapesLayer}
        />
      </Mapbox.ShapeSource>
    </View>
  );
}

function mapStateToProps(state) {
  return {
    visible: state.layerVisibility.routeShapes && !state.selectedArrival,
    routeShapes: state.routeShapes
  };
}

export default connect(mapStateToProps)(RouteShapesLayer);

const mapStyles = Mapbox.StyleSheet.create({
  routeShapesLayer: {
    lineColor: Mapbox.StyleSheet.identity("color"),
    lineOpacity: 1,
    lineWidth: 3
  }
});
