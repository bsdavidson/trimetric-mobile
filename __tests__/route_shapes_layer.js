import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import {mapStateToProps, RouteShapesLayer} from "../route_shapes_layer";

import {props} from "../__fixtures__/route_shapes_layer";

jest.mock("@mapbox/react-native-mapbox-gl");

it("renders routeShapesLayer", () => {
  const rslNull = renderer.create(<RouteShapesLayer />).toJSON();
  expect(rslNull).toMatchSnapshot();

  const rsl = renderer
    .create(
      <RouteShapesLayer
        visible={props.visible}
        routeShapes={props.routeShapes}
        selectedArrival={props.selectedArrival}
      />
    )
    .toJSON();
  expect(rsl).toMatchSnapshot();
});

it("has a mapStateToProps", () => {
  let props = mapStateToProps({
    layerVisibility: {
      routeShapes: []
    },
    selectedArrival: null
  });

  expect(props).toEqual({
    visible: true,
    routeShapes: {type: "FeatureCollection", features: []},
    selectedArrival: null
  });

  props = mapStateToProps({
    layerVisibility: {
      routeShapes: []
    },
    selectedArrival: {}
  });

  expect(props).toEqual({
    visible: false,
    routeShapes: {type: "FeatureCollection", features: []},
    selectedArrival: {}
  });
});
