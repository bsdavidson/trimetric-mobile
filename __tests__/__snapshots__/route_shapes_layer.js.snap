// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders routeShapesLayer 1`] = `null`;

exports[`renders routeShapesLayer 2`] = `
<ShapeSource
  id="route_shapes_source"
  shape={
    Object {
      "features": Array [
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -122.680501,
                  45.53012,
                ],
                Array [
                  -122.681354,
                  45.530079,
                ],
                Array [
                  -122.681105,
                  45.530084,
                ],
              ],
            ],
            "type": "MultiLineString",
          },
          "properties": Object {
            "color": "#CE0F69",
            "route_id": "194",
          },
          "type": "Feature",
        },
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -122.671375,
                  45.493955,
                ],
                Array [
                  -122.671107,
                  45.49395,
                ],
                Array [
                  -122.698618,
                  45.529861,
                ],
                Array [
                  -122.698646,
                  45.530509,
                ],
                Array [
                  -122.698649,
                  45.530616,
                ],
              ],
              Array [
                Array [
                  -122.685036,
                  45.514564,
                ],
                Array [
                  -122.685257,
                  45.514157,
                ],
                Array [
                  -122.685323,
                  45.514034,
                ],
                Array [
                  -122.685339,
                  45.514005,
                ],
              ],
            ],
            "type": "MultiLineString",
          },
          "properties": Object {
            "color": "#84BD00",
            "route_id": "193",
          },
          "type": "Feature",
        },
      ],
      "type": "FeatureCollection",
    }
  }
>
  <LineLayer
    belowLayerID="place-city-sm"
    id="route_shapes_layer"
    style={
      Array [
        Object {
          "lineColor": undefined,
          "lineOpacity": 1,
          "lineWidth": 3,
        },
        Object {
          "visibility": "visible",
        },
      ]
    }
  />
</ShapeSource>
`;
